File: hospitalManagement/Doctor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Doctor : Employee
    {


        private List<Patient> patientList;

        // Dynamic field
        // Properties

        internal List<Patient> PatientList { get => patientList; set => patientList = value; }

        // Constructors
        public Doctor() {
            patientList = new List<Patient>();
        }

        public Doctor(string id
        , string firstName
        , string lastName
        , int age
        , int gentle
        , string description
        , string address
        , string email
        , string phone
        , string facultyId
        , string departmentId
        , bool state
        , ComeAndLeaveDate admissionDates
        , Salary salaries
        ) : base(id
            , firstName
            , lastName
            , age
            , gentle
            , description
            , address
            , email
            , phone
            , facultyId
            , departmentId
            , state
            , admissionDates
            , salaries
            )
        {

            this.PatientList = patientList;
        }

        // Destructors
        ~Doctor() { }
        // Methods
        // in, Output
        public override void Input()
        {
            base.Input();
            //Console.WriteLine("The patient list: ");
            //Console.WriteLine("Number of patient in charge: ");
            //int i = Convert.ToInt32(Console.ReadLine());

            //for (int j = 0; j < i; j++)
            //{
            //    Patient patient = new Patient();
            //    patientList.Add(patient);
            //}
        }

        public override void Output()
        {
            base.Output();
            Console.WriteLine("List of patients in charge: ");
            patientList.ForEach(value => { value.Output(); Console.WriteLine(); });
        }
        // General method

        // Other method


        // Overriding

        public override void ExportBill()
        {
            base.ExportBill();
        }

        public override float CalcBill()
        {
            return base.CalcBill();
        }


    }
}

File: src/hospitalManagement/Doctor.java
package hospitalManagement;

import java.util.ArrayList;
import java.util.Scanner;

public class Doctor extends Employee{
       
        //Field
private ArrayList<Patient> patientList;



// Information field
//  Management field
static Scanner input = new Scanner(System.in);

// Dynamic field
// Properties

public ArrayList<Patient> getPatientList() {
    return patientList;
}
public void setPatientList(ArrayList<Patient> patientList) {
    this.patientList = patientList;
}


// Constructors
public Doctor()
{
    patientList = new ArrayList<Patient>();
}
public Doctor(String id
, String firstName
, String lastName
, int age
, int gentle
, String description
, String address
, String email
, String phone
, String facultyId
, String departmentId
, Boolean state
, ComeAndLeaveDate admissionDates
, Salary salaries
, ArrayList<Patient> patientList)
{
    super(id, firstName, lastName, age, gentle, description, address, email, phone, facultyId, departmentId, state, admissionDates, salaries);

    this.patientList = patientList;
}


// Methods
// in, output
public void Input()
{
    super.Input();
}

public  void Output()
{
    super.Output();
}
// General method
// Other method
// Overriding
public  void ExportBill()
{
    super.ExportBill();
}

public  float CalcBill()
{
    return super.CalcBill();
}
}
